test/integration/collection-view.spec.js

```diff
@@ -0,0 +1,139 @@
+const _ = Cypress._;
+const $ = Cypress.$;
+
+describe('CollectionView', () => {
+  let BbModel;
+  let BbCollection;
+  let CollectionView;
+  let MnView;
+
+  beforeEach(() => {
+    cy.server();
+
+    cy.visitMn(function({Marionette, Backbone}) {
+      BbModel = Backbone.Model;
+      BbCollection = Backbone.Collection;
+      MnView = Marionette.View;
+
+      const FooView = Marionette.View.extend({
+        template: _.template('Foo template content')
+      });
+
+      const BarView = Marionette.View.extend({
+        template: _.template('Bar template content')
+      });
+
+      CollectionView = Marionette.CollectionView.extend({
+        className: 'collection-view-container',
+        collection: new BbCollection(),
+        childView(item) {
+          if (item.get('isFoo')) {
+            return FooView;
+          } else {
+            return BarView;
+          }
+        }
+      });
+
+    });
+  });
+
+  describe('childView', () => {
+    let collectionView;
+    let fooModel;
+    let barModel;
+
+    beforeEach(() => {
+      const Model = BbModel.extend({
+        defaults: {
+          isFoo: false
+        }
+      });
+
+      fooModel = new Model({
+        isFoo: true
+      });
+
+      barModel = new Model({
+        isFoo: false
+      });
+
+      collectionView = new CollectionView();
+      collectionView.render();
+
+      $('#app-hook').html(collectionView.$el);
+    });
+
+    it('should be without childView\'s', function() {
+      cy.get('.collection-view-container').should('be.empty');
+    });
+
+    it('should show default childView', function() {
+      collectionView.collection.add(fooModel);
+
+      cy.get('#app-hook').should('contain', 'Foo template content');
+    });
+
+    it('should show a special childView', function() {
+      collectionView.collection.add(barModel);
+
+      cy.get('#app-hook').should('contain', 'Bar template content');
+    });
+  });
+
+  describe('emptyView', () => {
+    beforeEach(() => {
+      const EmptyView = MnView.extend({
+        className: 'empty-view',
+        template: _.template('Empty view content')
+      });
+      const CView = CollectionView.extend({
+        collection: new BbCollection([]),
+        emptyView: EmptyView
+      });
+      const collectionView = new CView();
+      collectionView.render();
+
+      $('#app-hook').html(collectionView.$el);
+    });
+
+    it('should have empty view element', () => {
+      cy.get('.collection-view-container').should($container => {
+        expect($container.find('.empty-view')).to.have.length(1);
+      });
+    });
+
+    it('should have empty view content', () => {
+      cy.get('.collection-view-container').should('contain', 'Empty view content');
```

Comment:
also I think after writing more complicated cypress tests, it's not possible to reasonably keep single assertions per `it`.  For performance and complexity reasons I think this test and the previous should be
`cy.get('.collection-view-container').find('.empty-view').should('contain', 'Empty view content');`
So many things include "implied asserts" such as `.find`, and Cypress will treat them fairly separately when running.